k8s/envoy-x509/README.md

```diff
@@ -219,93 +210,93 @@ We have deployed our services and also registered them in SPIRE. Now we can chec
+To update the Envoy configuration for the `backend` workload use the file `backend-envoy-configmap-update.yaml`:
 
-   ```console
-   $ kubectl delete pod $(kubectl get pods --selector=app=backend --output=jsonpath="{..metadata.name}")
-   ```
+```console
+   $ kubectl apply -f backend-envoy-configmap-update.yaml
+```
+
+Next, the `backend` pod needs to be restarted to pick up the new configurations:
+
+```console
+   $ kubectl scale deployment backend --replicas=0
+   $ kubectl scale deployment backend --replicas=1
+```
 
-Wait some seconds until de pods is running and ready before trying to hit the backend via frontend-2 service again.
-Once the pod is ready, refresh the browser on the correct url for the **frontend-2** service. In this case `35.222.190.182:3002`. As a result we can see that Envoy did not allow the request to get to the backend so the web page it is only showing the title without any account details.
+Wait some seconds for the deployment to propagate before trying to view the `frontend-2` service in your browser again.
+Once the pod is ready, refresh the browser using the correct URL for `frontend-2` service (e.g. `http://35.222.190.182:3002`). As a result, now Envoy does not allow the request to get to the `backend` service and account details are not shown in your browser.
 
 ![Frontend-2-no-details][frontend-2-view-no-details]
 
 [frontend-2-view-no-details]: images/frontend-2_view_no_details.png "Frontend-2 view no details account"
 
-
-On the other hand, you can check that the **frontend** service is still able to get a response from the **backend**. Refresh the browser at the correct url, in this case `35.222.164.221:3000`, and confirm that user details for _Jacob Marley_ are displayed.
+On the other hand, you can check that the `frontend` service is still able to get a response from the `backend`. Refresh the browser at the correct URL (e.g. `http://35.222.164.221:3000`) and confirm that account details are shown for _Jacob Marley_.
 
 
 # Cleanup
 
-When you are finished running this tutorial, you can use the following commands to remove all the resources used for Envoy integration and the SPIRE setup.
+When you are finished running this tutorial, you can use the following script to remove all the resources used for configuring Envoy to perform X.509 authentication on workload's behalf. This command will remove:
+   * - All resources created for the SPIRE - Envoy X.509 integration tutorial.
+   * - All deployments and configurations for the SPIRE agent, SPIRE server, and namespace.
 
-## Kubernetes Cleanup
+```console
+   $ bash scripts/clean-env.sh
+```
 
-Keep in mind that these commands will also remove the setup that you configured in the [Kubernetes Quickstart](/spire/try/getting-started-k8s/).
 
-1. Delete all resources created for this SPIRE - Envoy integration tutorial:
+# Extend the Scenario with a Role Based Access Control filter
```

Comment:
I think I'd put this new section before the cleanup section. Your thoughts? That is assuming that the cleanup scripts will also clean up the RBAC stuff.
```suggestion
# Extend the Scenario with a Role-Based Access Control Filter
```